source/admin_manual/list_of_events.rst

```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
+   :field syscw: Number of write syscalls
+                 (Skipped in non-Linux environments.)
 
-.. _event_mail_expunged:
 
-mail_expunged
-^^^^^^^^^^^^^
+Mailbox
+-------
 
-.. versionadded:: 2.3.15
+.. dovecot_core:event:: mail_expunged
+   :added: v2.3.15
+   :inherit: mailbox
 
-A mail was expunged from the mailbox. Note that this event inherits from
-mailbox, not mail.
+   :field uid: UID of the expunged mail.
 
-+---------------------+--------------------------------------------------------+
-| Field               | Description                                            |
-+=====================+========================================================+
-| uid                 | UID of the expunged mail.                              |
-+---------------------+--------------------------------------------------------+
+   A mail was expunged from the mailbox. Note that this event inherits from
+   mailbox, not mail.
 
-.. _event_mail:
 
 Mail
 ----
 
-+---------------------+------------------------------------------------------+
-| Field               | Description                                          |
-+=====================+======================================================+
-| Inherits from :ref:`event_mailbox`                                         |
-+---------------------+------------------------------------------------------+
-| seq                 | Mail sequence number                                 |
-+---------------------+------------------------------------------------------+
-| uid                 | Mail IMAP UID number                                 |
-+---------------------+------------------------------------------------------+
+.. dovecot_event:field_group:: mail
 
-.. _event_mail_opened:
+   :field seq: Mail sequence number.
+   :field uid: Mail IMAP UID number.
 
-mail_opened
-^^^^^^^^^^^
 
-.. versionadded:: 2.3.15
+.. dovecot_core:event:: mail_opened
+   :added: v2.3.15
+   :inherit: mail, mailbox
 
-A mail was opened e.g. for reading its body. Note that this event is not sent
-when mails' metadata is accessed, even if it causes opening the mail file.
+   :field reason: Reason why the mail was opened. (optional)
 
-+---------------------+--------------------------------------------------------+
-| Field               | Description                                            |
-+=====================+========================================================+
-| reason              | Reason why the mail was opened. (optional)             |
-+---------------------+--------------------------------------------------------+
+   A mail was opened e.g. for reading its body. Note that this event is not
+   sent when mails' metadata is accessed, even if it causes opening the mail
+   file.
 
-.. _event_mail_expunge_requested:
 
-mail_expunge_requested
-^^^^^^^^^^^^^^^^^^^^^^
+.. dovecot_core:event:: mail_expunge_requested
+   :added: v2.3.15
+   :inherit: mail, mailbox
```

Comment:
Does this need to list all inherited events? Couldn't it automatically go through the tree to list them all here by just specifying the first? Especially because now it stops at mailbox, why not continue to mail_user, mail_storage_user, ...

Reply:
Given my comments elsewhere, I agree that this page could be adopted and extended to work much better with inheritance.  But someone else needs to do that who has more insight to me (since I can't understand inheritance from the old documentation.)